### PR TITLE
Add extract-music

### PR DESCRIPTION
**Pull request by prophetl33t**
Allows for extraction of the most popular types of music (loading screen, main menu, some pve e.t.c.)

### CODE REVIEW

File: DataTool/DataModels/Unlock.cs [hunk 1]
```diff
@@ -264,7 +264,7 @@ public enum UnlockType {
         Icon,
         Spray,
         Emote,
-        VictoryPose,
+        VictoryPose, //wrestle with Jeff prepare for death
```
Comment (Js41637) [inline DataTool/DataModels/Unlock.cs:267]: wot

File: TankLib/STU/Types/Misc.cs [hunk 2]
```diff
@@ -3224,8 +3224,10 @@ public class STU_0FA0034C : STUInstance
         public uint m_wwiseEventIDStop;
         
         [STUField(0xF7805734, 164)] // size: 4
-        public uint m_F7805734;
-        
+        //Sound class.
+        //3851462253 is loading screen music
+        public uint m_soundClass;
+
```
Comment (Js41637) [inline TankLib/STU/Types/Misc.cs:3230]: this file cannot be edited, it is automatically generated, any changes made here will get undone when the file is regenerated causing errors in code that references it.

File: DataTool/ToolLogic/Extract/ExtractMusic.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    public class ExtractMusic : ITool
+    {
+
+        Dictionary<UInt32, string> music_types = new Dictionary<uint, string>
+        {   { 3851462253, "LoadingScreen" },
+            { 3043319348, "Retribution" },
+            { 935766596, "Uprising"},
+            { 1935561723, "StormRising1" },
+            { 3069540790, "StormRising2"},
```
Comment (Js41637) [inline DataTool/ToolLogic/Extract/ExtractMusic.cs:22]: There's probably like a hundred sounds related to storm rising, why the need for different numbers if they're for the same thing? You can reuse a name, it'll just be put in the same directory?

File: DataTool/ToolLogic/Extract/ExtractMusic.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            { 1287967541, "Junkenstein"},
+            { 3187645923, "PvP" },
+            { 3991942870, "Stinger" },
+            { 397648059,  "PvE"},
+            { 2930411284, "MainMenuTheme"},
+            { 251430280, "RoundNearEnd"}
```
Comment (Js41637) [inline DataTool/ToolLogic/Extract/ExtractMusic.cs:28]: What is PVP, PVE and Stinger type music?

**Review by Js41637 — CHANGES_REQUESTED** [review]

File: DataTool/ToolLogic/Extract/ExtractMusic.cs (hunk 3 above)
Comment (ZingBallyhoo) [reply DataTool/ToolLogic/Extract/ExtractMusic.cs:22]: also can we hex them :zingy:

File: DataTool/ToolLogic/Extract/ExtractMusic.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+using DataTool.Flag;
+using static DataTool.Program;
+using DataTool.SaveLogic;
+using TankLib.STU.Types;
+using static DataTool.Helper.STUHelper;
+using System.Collections.Generic;
+
+namespace DataTool.ToolLogic.Extract.Debug
+{
+    [Tool("extract-music", Description = "Extracts sound files which are identified as music.", CustomFlags = typeof(ExtractFlags), IsSensitive = true)]
```
Comment (Js41637) [inline DataTool/ToolLogic/Extract/ExtractMusic.cs:12]: this doesn't need to be sensitive :zingy:

**Review by Js41637 — CHANGES_REQUESTED** [review]

File: DataTool/ToolLogic/Extract/ExtractMusic.cs (hunk 4 above)
Comment (prophetl33t) [reply DataTool/ToolLogic/Extract/ExtractMusic.cs:28]: PvP is for music which can be used on any map in PvP (payload near destination, we are running out of time e.t.c.), same for pve. 
A stinger is a short clip of music. If name is confusing then i can change it.

File: DataTool/ToolLogic/Extract/ExtractMusic.cs (hunk 3 above)
Comment (prophetl33t) [reply DataTool/ToolLogic/Extract/ExtractMusic.cs]: > 
> 
> also can we hex them :zingy:

 ¯\_(ツ)_/¯

File: TankLib/STU/Types/Misc.cs (hunk 2 above)
Comment (yretenai) [reply TankLib/STU/Types/Misc.cs:3230]: also it's probably sound trigger, not sound class.

Comment (prophetl33t) [reply TankLib/STU/Types/Misc.cs:3230]: I think that m_wwiseEventIDPlay is a trigger :)

Comment (Js41637) [reply TankLib/STU/Types/Misc.cs:3230]: The name of the field is `m_wwiseBankID`, all the files with the same id also have the same STUSoundBank GUID. You can change it to that.

Comment (prophetl33t) [reply TankLib/STU/Types/Misc.cs:3230]: gotcha

File: TankLib/STU/Types/Misc.cs [hunk 6]
```diff
@@ -3224,9 +3224,8 @@ public class STU_0FA0034C : STUInstance
         public uint m_wwiseEventIDStop;
         
         [STUField(0xF7805734, 164)] // size: 4
-        //Sound class.
-        //3851462253 is loading screen music
-        public uint m_soundClass;
+
```
Comment (Js41637) [inline TankLib/STU/Types/Misc.cs:3227]: no new line :zingy:

File: DataTool/ToolLogic/Extract/ExtractMusic.cs [hunk 7]
```diff
@@ -49,7 +49,7 @@ public void ExtractType(ICLIFlags toolFlags)
             foreach (ulong @ulong in TrackedFiles[0x2C])
             {
                 STUSound music = GetInstance<STUSound>(@ulong);
-                var s_class = music.m_C32C2195.m_soundClass;
+                var s_class = music.m_C32C2195.m_wwiseBankID;
```
Comment (Js41637) [inline DataTool/ToolLogic/Extract/ExtractMusic.cs:52]: also can we reformat this file? it doesn't follow the style guide of the project which is essentially braces on same line. The formatting rules are in the project, you should just be able to format the file? Did it not auto attempt to follow the project style? It does for me in Visual Studio unless it's related to Resharper?

**Review by Js41637 — CHANGES_REQUESTED** [review]

**Comment by Js41637** [comment]
Other than that this is good, the music it finds is pretty good, lot of random sounds but it's better than nothing.

**Comment by Zekfad** [comment]
Will this effectively close #47?

**Comment by prophetl33t** [comment]
"wellllllllllllllllllllllllllllllllllllllllllllll
i guess" - JS

**Pull request merged**